pkg/R/context.R

```diff
@@ -19,6 +19,10 @@
 #'}
 
 textFile <- function(sc, path, minSplits = NULL) {
+  # Allow the user to have a more flexible definiton of the text file path
+  path <- path.expand(path)
+  if (!(grepl("\\\\", path) | grepl("/", path)))
```

Comment:
HI Shivaram,

 

normalizePath does look like a better choice for this.

 

Dan

 

From: Shivaram Venkataraman [mailto:notifications@github.com] 
Sent: Friday, January 02, 2015 11:56 AM
To: amplab-extras/SparkR-pkg
Cc: dputler
Subject: Re: [SparkR-pkg] Ability to more flexibly define the path to a text file (#119)

 

In 

> @@ -19,6 +19,10 @@
>  #'}
>  
>  textFile <- function(sc, path, minSplits = NULL) {
> +  # Allow the user to have a more flexible definiton of the text file path
> +  path <- path.expand(path)
> +  if (!(grepl("\\\\", path) | grepl("/", path)))

@dputler <https://github.com/dputler>  - No worries about the git client. I can fix things up during merge. One last thing I wanted to check with you was whether using normalizePath is better than using path.expand. From the documentation it looks like normalizePath calls path.expand first and then also handles things like ../ etc.

—
Reply to this email directly or view it on GitHub <https://github.com/amplab-extras/SparkR-pkg/pull/119/files#r22420752> .  <https://github.com/notifications/beacon/AIGvMldaurtD17bPejM55Zolu3vsjETaks5ndu9WgaJpZM4DJRxB.gif>